corehq/apps/app_manager/models.py

```diff
@@ -1519,18 +1520,48 @@ class MappingItem(DocumentSchema):
     value = DictProperty()
 
     @property
+    def treat_as_expression(self):
+        """
+        Returns if whether the key can be treated as a valid expression that can be included in
+        condition-predicate of an if-clause for e.g. if(<expression>, value, ...)
+        """
+        special_chars = '{}()[]=<>."\'/'
```

Comment:
Oh, I see, keys also allow spaces, for one. Still might be easier to limit this by specifying what's legal.